ludwig/config_validation/checks.py

```diff
@@ -409,3 +417,46 @@ def check_concat_combiner_requirements(config: "ModelConfig") -> None:  # noqa:
             "3D encoder output shapes, or 3) Remove features to ensure that output shapes from all encoders are the "
             "same dimension (all 2D or all 3D)."
         )
+
+
+@register_config_check
+def check_hyperopt_nested_parameter_dicts(config: "ModelConfig") -> None:  # noqa: F821
+    """Checks that all nested parameters in a hyperopt config exist."""
+    if config.hyperopt is None or "." not in config.hyperopt.parameters:
+        return
+
+    from ludwig.schema.hyperopt.utils import get_parameter_cls  # noqa: F401
+
+    invalid_fields = []
+    space = config.hyperopt.parameters["."]
+
+    def check_fields_exist(
+        parameter: dict, config: "BaseMarshmallowConfig", field_path: str = ""  # noqa: F821
+    ) -> bool:
+        """Recursively check config field existence."""
+        for k, v in parameter.items():
+            nested_path = f"{field_path}.{k}"
+            try:
+                if isinstance(v, dict):
+                    check_fields_exist(v, config.__getattribute__(k), field_path=nested_path)
+                else:
+                    config.__getattribute__(k)
+            except AttributeError:
+                invalid_fields.append(nested_path)
+
+    for category in space["categories"]:
+        check_fields_exist(category, config)
+
+    if len(invalid_fields) > 0:
+        raise ConfigValidationError(
+            "The following nested hyperparameters do not correspond to valid config fields: "
+            f"{','.join(invalid_fields)}. Check the Ludwig docs for the list of valid parameters."
+        )
+
+    try:
+        space_cls = get_parameter_cls("choice")
```

Comment:
nit/quibble: are we checking parameters here that are 'nested' or just ones that have categorical search spaces?

Reply:
Only nested spaces. The examples only show them as choice spaces, and I think expanding beyond choice or grid search is probably out of scope for this PR, so the check enforces that the space has to be choice.